### PR TITLE
Add an Android building image

### PR DESCRIPTION
**Pull request by directhex**
Closes https://github.com/dotnet/core-eng/issues/9431

### CODE REVIEW

File: manifest.json [hunk 1]
```diff
@@ -313,6 +313,21 @@
...
         },
+        {
+          "platforms": [
+            {
+              "dockerfile": "src/ubuntu/18.04/android",
+              "os": "linux",
+              "osVersion": "ubuntu18.04",
+              "tags": {
+                "ubuntu-18.04-android-$(System:DockerfileGitCommitSha)-$(System:TimeStamp)": {},
+                "ubuntu-18.04-android": {
```
Comment (mthalman) [inline manifest.json]: There's no need for this tag because there's no Dockerfile referencing it.

File: src/ubuntu/18.04/android/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,46 @@
+FROM mcr.microsoft.com/powershell:6.2.1-ubuntu-18.04
```
Comment (mthalman) [inline src/ubuntu/18.04/android/Dockerfile]: Much of this Dockerfile looks to be a copy of the coredeps Dockerfile.  Would it be better to change the base image to `mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-coredeps` and then just install the extra stuff not installed by that base image?

**Comment by akoeplinger** [comment]
@mthalman the `Build_Linux_amd64 src-ubuntu-18.04-coredeps-graph` build failed with:

```
Error processing tar file(exit status 1): write /crossrootfs/arm/var/cache/apt/srcpkgcache.bin: no space left on device
```

Is that related to the PR?

**Comment by directhex** [comment]
Unrelated to this PR.

**Pull request merged**